python/src/otel/otel_sdk/otel_wrapper.py

```diff
@@ -0,0 +1,79 @@
+import os
+import logging
+from opentelemetry import trace
+from importlib import import_module
+
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import (
+    SimpleExportSpanProcessor,
+    BatchExportSpanProcessor,
+)
+
+from opentelemetry.resource import AwsLambdaResourceDetector
+
+from opentelemetry.instrumentation.aws_lambda import AwsLambdaInstrumentor
+from pkg_resources import iter_entry_points
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# TODO: move to python-contrib
```

Comment:
Python sdk auto instrumentation will launch before Lambda python runtime, that means we have to some duties such as set Python system path, environment variables, etc. Any Lambda runtime change would break Python sdk auto instrumentation.
Another reason is for parity, wrapper is a universal solution for every Lambda language Runtime.

Reply:
Don't understand the launch order, isn't it exec wrapper -> wrapper (initialize SDK) -> user handler? We use normal SDK autoconfiguration for Java and it works well, I'd expect us to be able to take advantage of it for any language that supports env var configuration. For languages that don't, we may need to implement it.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/TracingRequestWrapperBase.java#L23

Reply:
In Lambda Python runtime the launch order is:
exec wrapper -> Lambda runtime(set system path) -> user handler.

OTel Python sdk auto-instrumentation is by fork, initialize OTel components  --fork to--> child process

If integrate python sdk auto-instrumentation into Lambda, the order would be:
exec wrapper -> OTel Python auto-instrumentation -> Lambda runtime -> user handler.
That means we need to handle system path in exec wrapper, else OTel Python auto-instrumentation does not work.

So, the wrapper solution is:
exec wrapper -> Lambda runtime -> wrapper -> user handler
It can avoid to touch the logic in Lambda runtime.

Reply:
Thanks for explaining the differences in the wrapper, including the forking. It would be good to figure out how that can be improved though - at the very least having some method that's exposed, indendent of the pythonpath setup, for processing variables so they don't have to be implemented here should be possible. Let's make sure we have a plan for making this as small as possible (Java exec wrapper is just 2 lines of code and no special SDK configuration, Python is more dynamic and should be even more possible :P)

Reply:
yes, I can raise PR to python sdk auto-instrumentation to simplify Lambda auto-instrumentation.